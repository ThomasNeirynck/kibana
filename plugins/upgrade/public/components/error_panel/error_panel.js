import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';


export function ErrorPanel({ children, className, title }) {
  const classes = classNames('kuiInfoPanel', 'kuiInfoPanel--error', className);

  return (
    <div className={ classes }>
      <div className="kuiInfoPanelHeader">
        <span className="kuiInfoPanelHeader__icon kuiIcon kuiIcon--error fa-warning" />
        <span className="kuiInfoPanelHeader__title">
          { title }
        </span>
      </div>

      <div className="kuiInfoPanelBody">
        <div className="kuiInfoPanelBody__message">
          { children }
        </div>
      </div>
    </div>
  );
}

ErrorPanel.propTypes = {
  children: PropTypes.node,
  classNames: PropTypes.string,
  title: PropTypes.string,
};

ErrorPanel.defaultProps = {
  children: null,
  classNames: '',
  title: 'An error occurred.',
};
