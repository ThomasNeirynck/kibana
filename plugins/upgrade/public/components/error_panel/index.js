export { ErrorPanel } from './error_panel';
