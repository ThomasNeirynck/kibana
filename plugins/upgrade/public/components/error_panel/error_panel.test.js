import { render } from 'enzyme';
import React from 'react';

import { ErrorPanel } from './error_panel';


describe('ErrorPanel', () => {
  test('renders without properties', () => {
    const component = (
      <ErrorPanel />
    );

    expect(render(component)).toMatchSnapshot();
  });

  test('renders the className property', () => {
    const component = (
      <ErrorPanel className="specificErrorPanel" />
    );

    expect(render(component)).toMatchSnapshot();
  });

  test('renders the title property', () => {
    const component = (
      <ErrorPanel title="A different title" />
    );

    expect(render(component)).toMatchSnapshot();
  });

  test('renders its children', () => {
    const component = (
      <ErrorPanel>
        Error message
      </ErrorPanel>
    );

    expect(render(component)).toMatchSnapshot();
  });
});
