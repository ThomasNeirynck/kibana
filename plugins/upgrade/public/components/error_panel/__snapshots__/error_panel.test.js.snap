// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPanel renders its children 1`] = `
<div
  class="kuiInfoPanel kuiInfoPanel--error"
>
  <div
    class="kuiInfoPanelHeader"
  >
    <span
      class="kuiInfoPanelHeader__icon kuiIcon kuiIcon--error fa-warning"
    />
    <span
      class="kuiInfoPanelHeader__title"
    >
      An error occurred.
    </span>
  </div>
  <div
    class="kuiInfoPanelBody"
  >
    <div
      class="kuiInfoPanelBody__message"
    >
      Error message
    </div>
  </div>
</div>
`;

exports[`ErrorPanel renders the className property 1`] = `
<div
  class="kuiInfoPanel kuiInfoPanel--error specificErrorPanel"
>
  <div
    class="kuiInfoPanelHeader"
  >
    <span
      class="kuiInfoPanelHeader__icon kuiIcon kuiIcon--error fa-warning"
    />
    <span
      class="kuiInfoPanelHeader__title"
    >
      An error occurred.
    </span>
  </div>
  <div
    class="kuiInfoPanelBody"
  >
    <div
      class="kuiInfoPanelBody__message"
    />
  </div>
</div>
`;

exports[`ErrorPanel renders the title property 1`] = `
<div
  class="kuiInfoPanel kuiInfoPanel--error"
>
  <div
    class="kuiInfoPanelHeader"
  >
    <span
      class="kuiInfoPanelHeader__icon kuiIcon kuiIcon--error fa-warning"
    />
    <span
      class="kuiInfoPanelHeader__title"
    >
      A different title
    </span>
  </div>
  <div
    class="kuiInfoPanelBody"
  >
    <div
      class="kuiInfoPanelBody__message"
    />
  </div>
</div>
`;

exports[`ErrorPanel renders without properties 1`] = `
<div
  class="kuiInfoPanel kuiInfoPanel--error"
>
  <div
    class="kuiInfoPanelHeader"
  >
    <span
      class="kuiInfoPanelHeader__icon kuiIcon kuiIcon--error fa-warning"
    />
    <span
      class="kuiInfoPanelHeader__title"
    >
      An error occurred.
    </span>
  </div>
  <div
    class="kuiInfoPanelBody"
  >
    <div
      class="kuiInfoPanelBody__message"
    />
  </div>
</div>
`;
